test/lib/pgBackRestTest/Module/Real/RealAllTest.pm

```diff
@@ -60,7 +60,7 @@ sub run
         {pg =>  '11', dst =>     'backup', tls => 1, stg => AZURE, enc => 0, cmp =>  ZST, rt => 2, bnd => 1, bi => 0},
```

Comment:
SFTP integration tests are only being run on RHEL. I tried to add one for Debian by updating the test for pg10:
```
{pg =>  '10', dst =>  'sftp-srvr', tls => 0, stg =>  SFTP, enc => 1, cmp =>   GZ, rt => 1, bnd => 0, bi => 0},
```
And I got this error:
```
ERROR: [101]: public key authentication failed: libssh2 error [-18]
HINT: libssh2 compiled against non-openssl libraries requires --repo-sftp-private-keyfile and --repo-sftp-public-keyfile to be provided
HINT: libssh2 versions before 1.9.0 expect a PEM format keypair, try ssh-keygen -m PEM -t rsa -P "" to generate the keypair
```

Reply:
Trying to replicate what you were getting, I just ran this:

sftp/test/test.pl --vm=u22 --module=real --log-level=info  --pg-version=10 --no-clean --vm-out

with a newly built u22 and got this:

2023-05-03 11:58:37.239 P00 ASSERT: [025]: expected 1 tests to run but 2 ran
    expected 1 tests to run but 2 ran
     at /home/sfrost/git/pg/pgbackrest/sftp/test/lib/pgBackRestTest/Common/RunTest.pm line 169.
        pgBackRestTest::Common::RunTest::process(undef, "u22", 0, "/home/sfrost/git/pg/pgbackrest/sftp", "/home/sfrost/git/pg/pgbackrest/test/test-0", "/home/sfrost/git/pg/pgbackrest/test/bin/u22/pgbackrest", "/home/sfrost/git/pg/pgbackrest/test/build/none/src/pgbackrest", "/usr/lib/postgresql/10/bin", ...) called at /home/sfrost/git/pg/pgbackrest/sftp/test/test.pl line 1187
        eval {...} called at /home/sfrost/git/pg/pgbackrest/sftp/test/test.pl line 1221

... but otherwise it all seemed to run successfully.  Happy to try and help debug the authentication issue but I haven't figured out how to reproduce it yet.  Are you sure you're on the latest commit which added in the ssh keys?  The auth error you see is exactly what I was getting before the key was added to the repo.

Reply:
Yes, I also made that change before running the test and getting the above error.

Reply:
Specifically:

diff --git a/test/lib/pgBackRestTest/Module/Real/RealAllTest.pm b/test/lib/pgBackRestTest/Module/Real/RealAllTest.pm
index 8f9de6aae..09a3628c3 100644
--- a/test/lib/pgBackRestTest/Module/Real/RealAllTest.pm
+++ b/test/lib/pgBackRestTest/Module/Real/RealAllTest.pm
@@ -57,6 +57,7 @@ sub run
         {pg => '9.5', dst =>     'backup', tls => 1, stg =>    S3, enc => 0, cmp =>  BZ2, rt => 1, bnd => 0, bi => 1},
         {pg => '9.6', dst =>     'backup', tls => 0, stg => POSIX, enc => 0, cmp => NONE, rt => 2, bnd => 1, bi => 1},
         {pg =>  '10', dst => 'db-standby', tls => 1, stg =>   GCS, enc => 1, cmp =>   GZ, rt => 2, bnd => 0, bi => 0},
+        {pg =>  '10', dst =>  'sftp-srvr', tls => 0, stg =>  SFTP, enc => 1, cmp =>   GZ, rt => 1, bnd => 0, bi => 0},
         {pg =>  '11', dst =>     'backup', tls => 1, stg => AZURE, enc => 0, cmp =>  ZST, rt => 2, bnd => 1, bi => 0},
         {pg =>  '12', dst =>     'backup', tls => 0, stg =>    S3, enc => 1, cmp =>  LZ4, rt => 1, bnd => 0, bi => 1},
         {pg =>  '13', dst => 'db-standby', tls => 1, stg =>   GCS, enc => 0, cmp =>  ZST, rt => 1, bnd => 1, bi => 1},

Reply:
ahhh, nevermind, needed to change it not add a line :)  Reproduced the auth error, looking at it now...

Reply:
I don't seem to be able to figure out how to effect the change into the sshd_config, but it looks like the issue, for me anyway, is that PubkeyAcceptedAlgorithms doesn't include ssh-rsa on u22 and the sshd_config needs to have it added like so:

PubkeyAcceptedAlgorithms=ssh-rsa

that then allows it to work.